src/main/java/uk/gov/hmcts/reform/unspec/service/flowstate/FlowPredicate.java

```diff
@@ -95,20 +139,27 @@
     public static final Predicate<CaseData> claimTakenOffline = caseData ->
         caseData.getCcdState() == PROCEEDS_IN_HERITAGE_SYSTEM;
 
+    public static final Predicate<CaseData> takenOfflineBySystem = caseData ->
+        caseData.getTakenOfflineDate() != null;
+
+    public static final Predicate<CaseData> takenOfflineByStaff = caseData ->
+        caseData.getTakenOfflineByStaffDate() != null;
+
     public static final Predicate<CaseData> caseProceedsInCaseman = caseData ->
```

Comment:
This one is not used anywhere now except in it's unit tests. Both of the predicates `takenOfflineByStaff` & `caseProceedsInCaseman` appears to be for the same event `CASE_PROCEEDS_IN_CASEMAN`